### DISCUSSION TITLE
Pwntools Tutorials 解题讨论

### DISCUSSION THREAD

**Discussion by wumingzhilian** [HUSTSeclab/software-security-dojo#18] [General]
我们在这里讨论 Pwntools Tutorials 模块中遇到的问题

**Comment by wyywwi** [comment]
Pwn最后一题（3.0）可以进行手动注入吗？尝试手动按照C程序中的指引进行输入修改笔记本达成bypass_me() 要求的条件，没有成功，不清楚是思路问题还是笔记本修改的问题。

**Reply by ZAIQuQ** [reply]
不知道手动输入会不会带换行符，我刚开始用`sendlineafter`就输入了换行符导致字符串比较不成功

**Comment by zeroyyx** [comment]
pwn2.0中在发送payload=asm('mov rax,0x12345678;',arch='amd64',os='linux')后能看见终端显示出print(open("/flag").read())\n'，但是后面显示error：Permission denied: '/flag'\n"，这是出了什么问题？

**Reply by wumingzhilian** [reply]
尝试选择正确的python去执行

**Reply by zeroyyx** [reply]
是尝试用python3运行自己编写的文件吗？发现用python执行和python3执行没有区别

**Reply by zeroyyx** [reply]
发现Permission denied: '/flag'\n"是由于运行python文件时通过p=process('sh') p.sendline('python /challenge/pwntools-tutorials-level2.0')运行文件时才会产生问题，后面修改运行方法就解决了

**Comment by mudongliang** [comment]
统一说明一下，level 1.1 这里有一个小bug，1.1 这个题号因为二进制未更新所以显示1.0，目前已经修改过来了
